modules/rtmessageadd.php

```diff
@@ -147,10 +157,21 @@
 				$message['destination'] = $queue['email'];
 			$recipients = $message['destination'];
 
-			if($message['userid'] && $addmsg)
-				$message['mailfrom'] = $queue['email'] ? $queue['email'] : $user['email'];
-			if($message['userid'] && !$addmsg)
-				$message['mailfrom'] = $user['email'] ? $user['email'] : $queue['email'];
+			$helpdesk_sender_email = ConfigHelper::getConfig('phpui.helpdesk_sender_email');
+			if(!empty($helpdesk_sender_email)) {
+				$mailfrom = $helpdesk_sender_email;
+
+				if($mailfrom == 'queue')
+					$mailfrom = $queue['email'];
+				elseif($mailfrom == 'user')
+					$mailfrom = $user['email'];
+			} else {
+				if($message['userid'] && $addmsg)
+					$message['mailfrom'] = $queue['email'] ? $queue['email'] : $user['email'];
+				if($message['userid'] && !$addmsg)
+					$message['mailfrom'] = $user['email'] ? $user['email'] : $queue['email'];
+			}
+			$message['mailfrom'] = $mailfrom;
```

Comment:
Look here please - you use $mailfrom which is not initialised if earlier else {} block. So $message['mailfrom'] will be unpredictable.

Reply:
should be fixed now :)